docs/data/data-grid/column-definition/column-definition.md

```diff
@@ -0,0 +1,228 @@
+---
+title: Data Grid - Column definition
+---
+
+# Data Grid - Column definition
+
+<p class="description">Define your columns.</p>
+
+The columns are defined with the `columns` prop which has the type `GridColDef[]`.
+
+`field` is the only required property since it's the column identifier. It's also used to match with `GridRowModel` values.
+
+```ts
+interface GridColDef {
+  /**
+   * The column identifier. It's used to match with [[GridRowModel]] values.
+   */
+  field: string;
+  …
+}
+```
+
+{{"demo": "BasicColumnsGrid.js", "bg": "inline"}}
+
+> ⚠️ The `columns` prop should keep the same reference between two renders.
+> The columns are designed to be definitions, to never change once the component is mounted.
+> Otherwise, you take the risk of losing elements like column width or order.
+> You can create the array outside the render function or memoize it.
+
+## Providing content
```

Comment:
Add a paragraph introducing this section? Same for "Rendering content".

Reply:
For "Rendering content" I just removed the `### Custom cell` title.
We now have

```
## Rendering content

// explains renderCell

### Styling cells

### Expand cell renderer
```

That way it goes from the most generic to the most advanced use case